src/fsharp/FSharp.Core/array.fs

```diff
@@ -57,7 +57,7 @@ namespace Microsoft.FSharp.Collections
         let create (count:int) (x:'T) =
             if count < 0 then invalidArg "count" (SR.GetString(SR.inputMustBeNonNegative))
             let array: 'T[] = Array.zeroCreateUnchecked count
-            for i = 0 to Operators.Checked.(-) count 1 do // use checked arithmetic here to satisfy FxCop
+            for i = 0 to Operators.Checked.(-) array.Length 1 do // use checked arithmetic here to satisfy FxCop
```

Comment:
Fxcop is a tool that checks libs if they conform to the framework
guidelines. I assume that at some point this was an Unchecked call and then
it got changed because some policy check failed.

But if it makes no perf hit then we should just use plain old + and -
operators and get rid of such questions.

Am 15.08.2016 15:01 schrieb "Jack Mott" <notifications@github.com>:

> In src/fsharp/FSharp.Core/array.fs
> <https://github.com/Microsoft/visualfsharp/pull/1420#discussion_r74757475>
> :
>
> > @@ -57,7 +57,7 @@ namespace Microsoft.FSharp.Collections
> >          let create (count:int) (x:'T) =
> >              if count < 0 then invalidArg "count" (SR.GetString(SR.inputMustBeNonNegative))
> >              let array: 'T[] = Array.zeroCreateUnchecked count
> > -            for i = 0 to Operators.Checked.(-) count 1 do // use checked arithmetic here to satisfy FxCop
> > +            for i = 0 to Operators.Checked.(-) array.Length 1 do // use checked arithmetic here to satisfy FxCop
>
> One of the functions that has it, has a comment about it being for FXCOP
> (I don't know what that is)
> I did test with and without it, and it made no difference on that
> function. I could test all of them though to be sure.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/Microsoft/visualfsharp/pull/1420/files/a177d8e52bcead89951983d8513dfa43a80c2bee#r74757475>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AADgNIs1GaQa21ZPMefpnusA9Bt6jKwAks5qgGMVgaJpZM4Jh_aU>
> .
>